gradle/wrapper/gradle-wrapper.properties

```diff
@@ -1,8 +1,8 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 # checksum was taken from https://gradle.org/release-checksums
-distributionSha256Sum=e111cb9948407e26351227dabce49822fb88c37ee72f1d1582a69c68af2e702f
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-bin.zip
+distributionSha256Sum=9631d53cf3e74bfa726893aee1f8994fee4e060c401335946dba2156f440f24c
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.6-bin.zip
```

Comment:
this seems unrelated to running this with JDK21. We currently can't upgrade Gradle due to an issue with RevAPI

Reply:
it is related. gradle 8.1.1 cannot be run on JDK 21

i am aware of the dependency, see https://github.com/apache/iceberg/pull/10474#issuecomment-2161508864